pkg/runctl/controller.go

```diff
@@ -244,37 +244,40 @@ func (c *Controller) syncHandler(key string) error {
 	}
 
 	runManager := c.createRunManager(pipelineRun, pipelineRunsConfig)
-	// Process pipeline run based on current state
-	switch state := pipelineRun.GetStatus().State; state {
-	// TODO fix #117
-	// Runs might be left in state `preparing` after a controller crash.
-	// Those must be recovered.
-	case api.StateUndefined:
-		c.metrics.CountStart()
+
+	if pipelineRun.GetStatus().State == api.StateUndefined {
 		if err = c.changeState(pipelineRun, api.StatePreparing); err != nil {
 			return err
 		}
+		c.metrics.CountStart()
+	}
+	// Process pipeline run based on current state
+	switch state := pipelineRun.GetStatus().State; state {
+	case api.StatePreparing:
```

Comment:
Do we need the `StatePreparing` at all - since we switch from `StateUndefined` to `StatePreparing` and immediately process it now. In `StateUndefined` we only switch to `StatePreparing` and do nothing else (besides metrics.CountStart()).

Reply:
I think this is a good way to count each start only once.

Reply:
In addition we have the timestamp of first pick-up in the status. This can be significantly differ from the creation timestamp.